Source/DOM classes/Unported or Partial DOM/SVGImageElement.m

```diff
@@ -105,7 +106,7 @@ - (CALayer *) newLayer
 	 */
 	AppleNativeImageRef image = [AppleNativeImage imageWithData:imageData];
 	
-    if( image == nil ) // NSData doesn't contain an imageformat Apple supports; might be an SVG instead
+    if( image == nil && imageURL ) // NSData doesn't contain an imageformat Apple supports; might be an SVG instead
```

Comment:
You may be right, but it's not necessary to continue if imageURL is nil due to parse fail. I'm not against the two methods you mentioned.